src/black/__init__.py

```diff
@@ -825,6 +853,9 @@ def format_file_contents(src_contents: str, *, fast: bool, mode: Mode) -> FileCo
     valid by calling :func:`assert_equivalent` and :func:`assert_stable` on it.
     `mode` is passed to :func:`format_str`.
     """
+    if mode.is_ipynb:
+        return format_ipynb_string(src_contents, mode=mode)
```

Comment:
This means we're skipping the equivalent and stable checks below for notebooks.

Reply:
Yes, you're right thanks. I've changed it so that, unless `--fast` is passed, the checks will be performed on a cell-by-cell basis, but after the cells have been masked (I don't see how to do it with the raw cells because they might not be parseable by `ast.parse` if they contain magics)